test/client_test.go

```diff
@@ -256,3 +257,42 @@ func TestCreateClientHttpConnectionCustomTransport(t *testing.T) {
 		t.Logf("Got server version %s", info)
 	}
 }
+
+// TestResponseHeader checks the Response.Header function.
+func TestResponseHeader(t *testing.T) {
+	c := createClientFromEnv(t, true)
+	ctx := context.Background()
+
+	version, err := c.Version(nil)
+	if err != nil {
+		t.Fatalf("Version failed: %s", describe(err))
+	}
+	isv33p := version.Version.CompareTo("3.3") >= 0
+	if !isv33p {
+		t.Skip("This test requires version 3.3")
+	} else {
+		var resp driver.Response
+		db := ensureDatabase(ctx, c, "_system", nil, t)
+		col := ensureCollection(ctx, db, "response_header_test", nil, t)
+
+		// `ETag` header must contain the `_rev` of the new document in quotes.
+		doc := map[string]string{
+			"Test":   "TestResponseHeader",
+			"Intent": "Check Response.Header",
+		}
+		meta, err := col.CreateDocument(driver.WithResponse(ctx, &resp), doc)
+		if err != nil {
+			t.Fatalf("CreateDocument failed: %s", describe(err))
+		}
+		expectedETag := strconv.Quote(meta.Rev)
```

Comment:
I would also add the "Etag" capitalization, although the generic handling should now cover it as well.

Reply:
Done